py/server/deephaven/json/__init__.py

```diff
@@ -0,0 +1,1221 @@
+
+    .. code-block:: python
+        some_method(string_val())
+
+    could be simplified to
+
+    .. code-block:: python
+        some_method(str)
+
+    Args:
+        allow_int (bool): if the string value is allowed to be a JSON integer type, default is False
+        allow_decimal (bool): if the string value is allowed to be a JSON decimal type, default is False
+        allow_bool (bool): if the string value is allowed to be a JSON boolean type, default is False
+        allow_missing (bool): if the double value is allowed to be missing, default is True
+        allow_null (bool): if the double value is allowed to be a JSON null type, default is True
+        on_missing (Optional[str]): the value to use when the JSON value is missing and allow_missing is True, default is None.
+        on_null (Optional[str]): the value to use when the JSON value is null and allow_null is True, default is None.
+
+    Returns:
+        the String value
+    """
+    builder = _JStringValue.builder()
+    _build(
+        builder,
+        allow_missing,
+        allow_null,
+        allow_string=True,
+        allow_int=allow_int,
+        allow_decimal=allow_decimal,
+        allow_bool=allow_bool,
+    )
+    if on_null:
+        builder.onNull(on_null)
+    if on_missing:
+        builder.onMissing(on_missing)
+    return JsonValue(builder.build())
+
+
+# TODO(deephaven-core#5269): Create deephaven.time time-type aliases
+def instant_val(
+    allow_missing: bool = True,
+    allow_null: bool = True,
+    number_format: Literal[None, "s", "ms", "us", "ns"] = None,
+    allow_decimal: bool = False,
+    on_missing: Optional[Any] = None,
+    on_null: Optional[Any] = None,
+) -> JsonValue:
+    """Creates an Instant value. For example, the JSON string
+
+    .. code-block:: json
+        "2009-02-13T23:31:30.123456789Z"
+
+    might be modelled as the Instant type
+
+    .. code-block:: python
+        instant_val()
+
+    In another example, the JSON decimal
+
+    .. code-block:: json
+        1234567890.123456789
+
+    might be modelled as the Instant type
+
+    .. code-block:: python
+        instant_val(number_format="s", allow_decimal=True)
+
+    In contexts where the user needs to create a JsonValueType and isn't changing any default values, the user can
+    simplify by using the python datetime type. For example,
+
+    .. code-block:: python
+        some_method(instant_val())
+
+    could be simplified to
+
+    .. code-block:: python
+        some_method(datetime)
+
+    Args:
+        allow_missing (bool): if the Instant value is allowed to be missing, default is True
+        allow_null (bool): if the Instant value is allowed to be a JSON null type, default is True
+        number_format (Literal[None, "s", "ms", "us", "ns"]): when set, signifies that a JSON numeric type is expected.
+            "s" is for seconds, "ms" is for milliseconds, "us" is for microseconds, and "ns" is for nanoseconds.
```

Comment:
from the epoch?

Reply:
Shoiuld this also indicate that the default `None` expects a string formatted as XYZ?